src/parser.ts

```diff
@@ -414,30 +425,74 @@ function parseAnySwitchFromToolArguments(args: string, excludeArgs: string[]): s
+        // A fragment like "-sw1,-sw2,-sw3" gets split by comma and a fragment like
+        // "-SwDef=Val" is split by equal. Opened GitHub issue
+        // https://github.com/microsoft/vscode-makefile-tools/issues/149.
+        // For now, these scenarios don't happen on windows (the comma syntax is linux only
+        // and the equal syntax is encountered for defines and c/c++ standard which are parsed
+        // separately, not via this script invocation).
+        parseCompilerArgsScriptContent = `@echo off\r\n`;
+        parseCompilerArgsScriptContent += `for %%i in (%*) do echo %%i \r\n`;
+    } else {
+        parseCompilerArgsScriptContent = `#!/bin/bash \n`;
+        parseCompilerArgsScriptContent += `while (( "$#" )); do \n`;
+        parseCompilerArgsScriptContent += `  echo $1 \n`;
+        parseCompilerArgsScriptContent += `  shift \n`;
+        parseCompilerArgsScriptContent += `done \n`;
+    }
+
+    // Create the script only when not found. The extension makes sure to delete it regularly
+    // (at project load time).
+    if (!util.checkFileExistsSync(parseCompilerArgsScriptFile)) {
+        util.writeFile(parseCompilerArgsScriptFile, parseCompilerArgsScriptContent);
+    }
+
+    let scriptArgs: string[] = [];
+    let runCommand: string;
+    if (process.platform === 'win32') {
+        runCommand = "cmd";
+        scriptArgs.push("/c");
+        scriptArgs.push(parseCompilerArgsScriptFile);
+    } else {
+        runCommand = "/bin/bash";
+        scriptArgs.push("-c");
+        scriptArgs.push(`"source ${parseCompilerArgsScriptFile}"`);
+    }
 
-            swiValues.forEach(value => {
-                // The end of the current switch value
-                let index3: number = partialArgs.indexOf(value) + value.length;
-                let finalSwitch: string = partialArgs.substring(0, index3);
+    scriptArgs.push(compilerArgRegions);
 
-                finalSwitch = finalSwitch.trim();
-                switches.push(finalSwitch);
+    try {
+        let stdout: any = (result: string): void => {
+            let results: string[] = result.replace(/\r\n/mg, "\n").split("\n");
+            // In case of concatenated separators, the shell sees different empty arguments
+            // which we can remove (most common is more spaces not being seen as a single space).
+            results.forEach(res => {
+                if (res !== "") {
+                    switches.push(res.trim());
+                }
             });
-        }
+        };
 
-        match1 = match2;
+        let stderr: any = (result: string): void => {
+            logger.messageNoCR(`Error while running the compiler args parser script '${parseCompilerArgsScriptFile}'` +
+                               `for regions "${compilerArgRegions}": "${result}"`, "Normal");
+        };
+
+        const result: util.SpawnProcessResult = await util.spawnChildProcess(runCommand, scriptArgs, vscode.workspace.rootPath || "", stdout, stderr);
```

Comment:
`vscode.workspace.rootPath` is deprecated and should not be used.  We'll need to get this value out of `vscode.workspace.workspaceFolders`

Reply:
Right, done.